### PR TITLE
Fix string ledger_index

### LINKED ISSUE #1014: ledger_index  type in the ledger response of clio differs with rippled.
request:
`{"method": "ledger", "params": [{"transactions": false, "expand": false, "owner_funds": false, "ledger_index": "validated", "api_version": 2, "ledger_hash": "46BF445D434BE4332D5C466CDCE624B34E04AD9105D18C3F5788972A5AF7416C", "Fee": 20}]}
`

Rippled:
`{
  'account_hash': '41EDE981EBDD8469C2B6194CE2CACABF1235B2CE9BC4F73486C7A7CD46F32B82',
  'close_flags': 0,
  'close_time': 754880530,
  'close_time_human': '2023-Dec-03 01:02:10.000000000 UTC',
  'close_time_iso': '2023-12-03T01:02:10Z',
  'close_time_resolution': 10,
  'closed': True,
  'ledger_hash': '46BF445D434BE4332D5C466CDCE624B34E04AD9105D18C3F5788972A5AF7416C',
  'ledger_index': 2151309,
  'parent_close_time': 754880522,
  'parent_hash': 'DFAA20C7B700353581BB96E58B85DFF5F4DB25B015A5D6A185DD88FAE74691C7',
  'total_coins': '99999027323338605',
  'transaction_hash': '2A073E7C13C5A61C4BBCFF062F1D92760195FEBC59FAD34EDFB4D5F5DDED10B9'
}`

Clio:
`{
  'account_hash': '41EDE981EBDD8469C2B6194CE2CACABF1235B2CE9BC4F73486C7A7CD46F32B82',
  'close_flags': 0,
  'close_time': 754880530,
  'close_time_human': '2023-Dec-03 01:02:10.000000000 UTC',
  'close_time_resolution': 10,
  'close_time_iso': '2023-12-03T01:02:10Z',
  'ledger_hash': '46BF445D434BE4332D5C466CDCE624B34E04AD9105D18C3F5788972A5AF7416C',
  'ledger_index': '2151309',
  'parent_close_time': 754880522,
  'parent_hash': 'DFAA20C7B700353581BB96E58B85DFF5F4DB25B015A5D6A185DD88FAE74691C7',
  'total_coins': '99999027323338605',
  'transaction_hash': '2A073E7C13C5A61C4BBCFF062F1D92760195FEBC59FAD34EDFB4D5F5DDED10B9',
  'closed': True
}`

### PR DESCRIPTION
**Pull request by cindyyan317**
When api_version is 2, ledger_index should be always int.
Fix #1014

### CODE REVIEW

File: src/rpc/RPCHelpers.cpp [hunk 1]
```diff
@@ -372,7 +372,11 @@ toJson(ripple::LedgerHeader const& lgrInfo, bool const binary)
         header[JS(close_time_resolution)] = lgrInfo.closeTimeResolution.count();
         header[JS(close_time_iso)] = ripple::to_string_iso(lgrInfo.closeTime);
         header[JS(ledger_hash)] = ripple::strHex(lgrInfo.hash);
-        header[JS(ledger_index)] = std::to_string(lgrInfo.seq);
+        if (apiVersion < 2) {
```
Comment (godexsoft) [inline src/rpc/RPCHelpers.cpp]: Let's use `2u` so we don't get a warning

File: unittests/rpc/RPCHelpersTests.cpp [hunk 2]
```diff
@@ -443,7 +443,7 @@ TEST_F(RPCHelpersTest, DeliverMaxAliasV2)
 TEST_F(RPCHelpersTest, LedgerHeaderJson)
 {
     auto const ledgerHeader = CreateLedgerInfo(INDEX1, 30);
-    auto const binJson = toJson(ledgerHeader, true);
+    auto const binJson = toJson(ledgerHeader, true, 1);
```
Comment (godexsoft) [inline unittests/rpc/RPCHelpersTests.cpp]: Consider using `1u` here and in other places below where `toJson` is used

File: unittests/rpc/handlers/LedgerDataTests.cpp [hunk 3]
```diff
@@ -303,6 +300,62 @@ TEST_F(RPCLedgerDataHandlerTest, NoMarker)
...
+
+    EXPECT_CALL(*rawBackendPtr, doFetchLedgerObjects).WillOnce(Return(bbs));
+
+    runSpawn([&, this](auto yield) {
+        auto const handler = AnyHandler{LedgerDataHandler{mockBackendPtr}};
+        auto const req = json::parse(R"({"limit":10})");
+        auto output = handler.process(req, Context{.yield = yield, .apiVersion = 2});
+        ASSERT_TRUE(output);
+        EXPECT_TRUE(output->as_object().contains("ledger"));
+        //"close_time_human" 's format depends on platform, might be sightly different
```
Comment (godexsoft) [inline unittests/rpc/handlers/LedgerDataTests.cpp]: Please add some space before the comment line and let's use spaces and capitalisation to make it more readable. Something like this:

// Note: the format of "close_time_human" depends on the platform and might differ per platform. It is however guaranteed to be consistent on the same platform.

File: src/rpc/RPCHelpers.cpp (hunk 1 above)
Comment (godexsoft) [reply src/rpc/RPCHelpers.cpp]: I'd also maybe move this if/else block to the bottom and have a space between the block and the rest of the code.

**Review by godexsoft — CHANGES_REQUESTED** [review]
Looks good, just a few small comments regarding formatting and types 👍

**Review by kuznetsss — APPROVED** [review]
LGTM

**Review by godexsoft — APPROVED** [review]
LGTM!

**Pull request merged**